src/fshtypes/FshCodeSystem.ts

```diff
@@ -26,6 +28,24 @@ export class FshCodeSystem extends FshEntity {
     if (this.rules.find(rule => rule instanceof ConceptRule && rule.code == newConcept.code)) {
       throw new CodeSystemDuplicateCodeError(this.id, newConcept.code);
     }
+    // check the hierarchy, if applicable
+    // for each predecessor element in the new concept's hierarchy, we should be able to find a rule that
+    // 1. defines a concept
+    // 2. with the predecessor's code
+    // 3. and a hierarchy equal to everything before it in the new concept's hierarchy
+    newConcept.hierarchy.forEach((predecessor, i) => {
+      if (
+        !this.rules.some(rule => {
+          return (
+            rule instanceof ConceptRule &&
+            rule.code === predecessor &&
+            isEqual(rule.hierarchy, newConcept.hierarchy.slice(0, i))
+          );
+        })
+      ) {
+        throw new CodeSystemIncorrectHierarchyError(this.id, newConcept.code);
```

Comment:
I'm not sure we can safely do this check here, what about RuleSets:
```
CodeSystem: MyCS
* insert MyRS
* #code1 #code2 "foo"

RuleSet: MyRS
* #code1
```
Can we do this on the export side instead?

Reply:
So, this is something that I also realized recently: InsertRules can make all kinds of things happen on CodeSystems that shouldn't be able to happen. I opened [a ticket](https://standardhealthrecord.atlassian.net/browse/CIMPL-726) for this, and I think your comment here would be in scope for that task. I think leaving the implementation like this for this PR is fine, as it is consistent with the existing implementation, and they will both be addressed by that new task.